### PR TITLE
reconnect: Check wheather internet is working before reloading.

### PR DESCRIPTION
**Pull request by priyank-p**
When online event is triggered check weather the internet is atually working or not.
Commonly on windows it turns out that internet takes couple of seconds to boot up after
connecting to internet or in some cases this might be they have to sign in to internet service
portal in order to access internet.

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "electron-spellchecker": "1.1.2",
     "electron-updater": "2.18.2",
     "electron-window-state": "4.1.1",
+    "is-online": "^7.0.0",
```
Comment (akashnimare) [inline app/package.json]: Should bt the exact working version here.

Comment (priyank-p) [reply app/package.json]: Done

File: app/renderer/js/main.js [hunk 2]
```diff
@@ -488,9 +489,14 @@ class ServerManagerView {
 
 window.onload = () => {
 	const serverManagerView = new ServerManagerView();
+	const reconnectUtil = new ReconnectUtil(serverManagerView);
 	serverManagerView.init();
 	window.addEventListener('online', () => {
-		serverManagerView.reloadView();
+		reconnectUtil.pollInternetAndReload();
```
Comment (akashnimare) [inline app/renderer/js/main.js:495]: Is there a need for adding a console log, just to make the debugging better (for issues like https://github.com/zulip/zulip-electron/issues/312) -
`console.log('You're back online.')`
`console.log('No internet connection, you are offline.')`

Comment (priyank-p) [reply app/renderer/js/main.js:495]: Ah yeah, so its easier to figure out.

File: app/renderer/js/utils/reconnect-util.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	_checkAndReload() {
+		return new Promise(resolve => {
+			if (!this.alreadyReloaded) { // eslint-disable-line no-negated-condition
+				isOnline()
+				.then(online => {
+					if (online) {
+						if (!this.alreadyReloaded) {
+							this.serverManagerView.reloadView();
+						}
+						console.log("You're back online.");
```
Comment (akashnimare) [inline app/renderer/js/utils/reconnect-util.js]: `console.log('You're back online.');`

Comment (akashnimare) [reply app/renderer/js/utils/reconnect-util.js]: @cPhost this.

Comment (priyank-p) [reply app/renderer/js/utils/reconnect-util.js]: Ah. i forgot to `git add` this :)

Comment (priyank-p) [reply app/renderer/js/utils/reconnect-util.js]: I though if you had an `'` inside a string that needs to escaped we can use double quotes.

**Pull request merged**